### PR TITLE
feat(button): added feedback icon

### PR DESCRIPTION
**Pull request by captain-Akshay**
**Notes for Reviewers**

This PR fixes #

**[Signed commits](../blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**

- [x] Theme for dark mode is in Progress

- [X] Yes, I signed my commits.

[Screencast from 2024-03-01 18-37-07.webm](https://github.com/layer5io/sistent/assets/59491379/824eb7a2-d183-4a2d-af0d-d6109ab44e54)

### CODE REVIEW

File: src/custom/Feedback/FeedbackButton.tsx [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+                    label={''}
+                    style={{
+                      color: 'white',
+                      fontSize: '2px',
+                      marginLeft: '0px',
+                      marginRight: '5px'
+                    }}
+                    control={
+                      <Checkbox
+                        style={{ color: '#F6F8F8' }}
```
Comment (aabidsofi19) [inline src/custom/Feedback/FeedbackButton.tsx:170]: @captain-Akshay  use directly from theme

File: src/custom/Feedback/FeedbackButton.tsx [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+                        checked={isChecked}
+                        onChange={handleCheckboxChange}
+                      />
+                    }
+                  />
+                  <Typography style={{ fontSize: '0.8rem', marginRight: '10px', color: 'white' }}>
```
Comment (aabidsofi19) [inline src/custom/Feedback/FeedbackButton.tsx:176]: use from theme

File: src/custom/Feedback/FeedbackButton.tsx [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+                    We may email you for more information or updates
+                  </Typography>
+                </div>
+                <FeedbackSubmitButton
+                  type="submit"
+                  disabled={!(messageValue && isChecked)}
+                  style={{ backgroundColor: !(messageValue && isChecked) ? '#b0bec5' : '#00B39F' }}
```
Comment (aabidsofi19) [inline src/custom/Feedback/FeedbackButton.tsx:183]: theme

File: src/custom/Feedback/FeedbackButton.tsx [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+                  onClick={handleSubmit}
+                >
+                  Send
+                </FeedbackSubmitButton>
+              </div>
+            }
+            leftHeaderIcon={<FeedbackIcon width={'24'} height={'24'} fill="#F6F8F8" />}
```
Comment (aabidsofi19) [inline src/custom/Feedback/FeedbackButton.tsx:190]: theme

File: src/custom/Feedback/FeedbackButton.tsx [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+                  {feedbackData?.map((item) => (
+                    <FeedbackOptionButton
+                      key={item.label}
+                      type="button"
+                      onClick={() => {
+                        setCategory(item);
+                      }}
+                      style={{
+                        borderBottom: category?.label === item.label ? '2px solid #00B39F' : '',
+                        background: category?.label === item.label ? '#fff' : '',
```
Comment (aabidsofi19) [inline src/custom/Feedback/FeedbackButton.tsx:212]: themeeeeee

**Review by aabidsofi19 — CHANGES_REQUESTED** [review]
Why are colors hardcoded , why is sistent theme not used ?

**Comment by ritiksaxena124** [comment]
Use this icon for Meet Request @captain-Akshay 
Meet Request Icon

File: src/custom/Feedback/FeedbackButton.tsx (hunk 1 above)
Comment (captain-Akshay) [reply src/custom/Feedback/FeedbackButton.tsx:170]: yes that's what i m working on.

File: src/custom/Feedback/FeedbackButton.tsx (hunk 5 above)
Comment (captain-Akshay) [reply src/custom/Feedback/FeedbackButton.tsx:212]: yes yes i wrote that in description that dark mode is in progress

**Review by leecalcote — APPROVED** [review]

File: src/custom/Feedback/FeedbackButton.tsx [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+  {
+    icon: <IdeaIcon />,
+    label: 'Suggestion',
+    placeholder: 'I have a suggestion about...',
+    isTextInput: true
+  },
+  {
+    icon: <CalenderIcon />,
+    label: 'Meet Request',
+    placeholder: 'You have a suggestion3?',
```
Comment (leecalcote) [inline src/custom/Feedback/FeedbackButton.tsx:70]: @captain-Akshay what is this? .... I'll tell you: it's not what I requested. Revisit the mockup.

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: src/custom/Feedback/FeedbackButton.tsx [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+                    We may email you for more information or updates
+                  </Typography>
+                </div>
+                <FeedbackSubmitButton
+                  type="submit"
+                  disabled={!(messageValue && isChecked)}
+                  isOpen={!(messageValue && isChecked)}
+                  onClick={handleSubmit}
+                >
+                  Send
```
Comment (leecalcote) [inline src/custom/Feedback/FeedbackButton.tsx:187]: Fix this, please. My mockup and my feedback both have not been implemented here.

File: src/custom/Feedback/FeedbackButton.tsx [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+                </FeedbackSubmitButton>
+              </div>
+            }
+            leftHeaderIcon={<FeedbackIcon fill="#FBFBFB" />}
```
Comment (leecalcote) [inline src/custom/Feedback/FeedbackButton.tsx:191]: Use theme references.

File: src/icons/Success/SuccessIcon.tsx [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+  return (
+    <svg
+      width={width}
+      height={height}
+      version="1.1"
+      xmlns="http://www.w3.org/2000/svg"
+      viewBox="0 0 50 50"
+      {...props}
+    >
+      <circle fill="#25AE88" cx="25" cy="25" r="25" />
```
Comment (leecalcote) [inline src/icons/Success/SuccessIcon.tsx:14]: theme reference, so that light and dark mode works.

File: src/icons/Question/QuestionIcon.tsx [hunk 10]
```diff
@@ -0,0 +1,19 @@
+import { FC } from 'react';
+import { IconProps } from '../types';
+
+export const QuestionIcon: FC<IconProps> = ({ width, height, ...props }) => (
+  <svg
+    width={width}
+    height={height}
+    xmlns="http://www.w3.org/2000/svg"
+    viewBox="0 0 24 24"
```
Comment (leecalcote) [inline src/icons/Question/QuestionIcon.tsx:9]: please add `fill` as a prop and white as default.

File: src/icons/Idea/IdeaIcon.tsx [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+        d="M9.87735 18.4805H5.11544C4.9183 18.4805 4.7583 18.6415 4.7583 18.8405V19.0805C4.7583 19.2794 4.9183 19.4405 5.11544 19.4405H9.87735C10.0745 19.4405 10.2345 19.2794 10.2345 19.0805V18.8405C10.2345 18.6415 10.0745 18.4805 9.87735 18.4805Z"
+        fill="#00B39F"
+      />
+      <path
+        d="M9.87735 19.9199H5.11544C4.9183 19.9199 4.7583 20.081 4.7583 20.2799V20.5199C4.7583 20.7189 4.9183 20.8799 5.11544 20.8799H9.87735C10.0745 20.8799 10.2345 20.7189 10.2345 20.5199V20.2799C10.2345 20.081 10.0745 19.9199 9.87735 19.9199Z"
+        fill="#00D3A9"
+      />
+      <path
+        d="M9.87735 21.3604H5.11544C4.9183 21.3604 4.7583 21.5214 4.7583 21.7204V21.9604C4.7583 22.1593 4.9183 22.3204 5.11544 22.3204H9.87735C10.0745 22.3204 10.2345 22.1593 10.2345 21.9604V21.7204C10.2345 21.5214 10.0745 21.3604 9.87735 21.3604Z"
+        fill="#00B39F"
```
Comment (leecalcote) [inline src/icons/Idea/IdeaIcon.tsx:26]: use theme reference and fill prop

File: src/icons/Feedback/FeedbackIcon.tsx [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+const FeedbackIcon: FC<IconProps> = ({
+  width = DEFAULT_WIDTH,
+  height = DEFAULT_HEIGHT,
+  ...props
+}) => {
+  return (
+    <svg
+      width={width}
+      height={height}
+      version="1.1"
```
Comment (leecalcote) [inline src/icons/Feedback/FeedbackIcon.tsx:14]: add fill

File: src/icons/Close/CloseIcon.tsx [hunk 13]
```diff
@@ -4,6 +4,7 @@ import { IconProps } from '../types';
 export const CloseIcon = ({
   width = DEFAULT_WIDTH,
   height = DEFAULT_HEIGHT,
+  fill = '#F6F8F8',
```
Comment (leecalcote) [inline src/icons/Close/CloseIcon.tsx:7]: don't hardcode

File: src/icons/Calender/CalenderIcon.tsx [hunk 14]
```diff
@@ -0,0 +1,25 @@
+import { FC } from 'react';
+import { IconProps } from '../types';
+
+const CalenderIcon: FC<IconProps> = ({ width, height, ...props }) => {
+  return (
+    <svg
+      width={width}
+      height={height}
+      version="1.1"
```
Comment (leecalcote) [inline src/icons/Calender/CalenderIcon.tsx:9]: add fill

File: src/icons/Calender/CalenderIcon.tsx [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+      xmlns="http://www.w3.org/2000/svg"
+      viewBox="0 0 19 24"
+      {...props}
+    >
+      <path d="M14 9H4V11H14V9Z" />
+      <path d="M2 18V7H16V13H18V4C18 3.46957 17.7893 2.96086 17.4142 2.58579C17.0391 2.21071 16.5304 2 16 2H15V0H13V2H5V0H3V2H2C1.46957 2 0.960859 2.21071 0.585786 2.58579C0.210714 2.96086 0 3.46957 0 4L0 18C0 18.5304 0.210714 19.0391 0.585786 19.4142C0.960859 19.7893 1.46957 20 2 20H12V18H2Z" />
+      <path d="M11 13H4V15H11V13Z" />
+      <path
+        d="M15.256 20V17.744H13V16.244H15.256V14H16.756V16.244H19V17.744H16.756V20H15.256Z"
+        fill="#00B39F"
```
Comment (leecalcote) [inline src/icons/Calender/CalenderIcon.tsx:19]: stop repeating yourself. use the theme.

File: src/custom/Feedback/style.tsx [hunk 16]
```diff
@@ -0,0 +1,158 @@
...
+  marginBottom: '0.5rem',
+  display: 'flex',
+  borderRadius: '0px',
+  flexDirection: 'column',
+  borderBottom: isOpen ? '2px solid #00B39F' : '',
+  background: isOpen
+    ? theme.palette.mode === 'dark'
+      ? '#000'
+      : '#FBFBFB'
+    : theme.palette.mode === 'dark'
```
Comment (leecalcote) [inline src/custom/Feedback/style.tsx:111]: @nebula-aac are these conditionals necessary? Can't the current mode be specified centrally?

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: src/custom/Feedback/FeedbackButton.tsx [hunk 17]
```diff
@@ -0,0 +1,231 @@
...
+    setSubmitted(false);
+    setMessageValue('');
+  };
+
+  const handleSubmit = () => {
+    setSubmitted(true);
+    if (messageValue && isChecked) {
+      onSubmit({ label: category?.label || '', message: messageValue });
+    }
+    setTimeout(() => {
```
Comment (aabidsofi19) [inline src/custom/Feedback/FeedbackButton.tsx:128]: Ouch .. timeouts  . @captain-Akshay  what is the state policy against timeouts here

**Review by aabidsofi19 — CHANGES_REQUESTED** [review]

File: src/custom/Feedback/FeedbackButton.tsx (hunk 6 above)
Comment (captain-Akshay) [reply src/custom/Feedback/FeedbackButton.tsx:72]: that placeholder is not used anywhere it's just a prop the whole component doesn't render it placeholder unless text field true and false is specified.

Comment (captain-Akshay) [reply src/custom/Feedback/FeedbackButton.tsx:72]: Though removed it now

**Review by sudhanshutech — COMMENTED** [review]
@captain-Akshay have you exported customtexttooltip from index file otherwise it will not work when try to import in other project

**Pull request merged**